pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -152,32 +138,6 @@ func (isvc *InferenceService) assignSKLearnRuntime() {
 	if isvc.Spec.Predictor.SKLearn.ProtocolVersion != nil &&
 		constants.ProtocolV2 == *isvc.Spec.Predictor.SKLearn.ProtocolVersion {
 		runtime = constants.MLServer
```

Comment:
Is this check still needed as we do auto selection based on `protocolVersion` now?